app/controllers/socure_webhook_controller.rb

```diff
@@ -8,11 +8,15 @@ class SocureWebhookController < ApplicationController
   check_or_render_not_found -> { IdentityConfig.store.socure_webhook_enabled }
 
   def create
-    if token_valid?
-      render json: { message: 'Secret token is valid.' }
-    else
-      render status: :unauthorized, json: { message: 'Invalid secret token.' }
+    unless token_valid?
+      return render(status: :unauthorized, json: { message: 'Invalid secret token.' })
     end
+    if socure_params[:event].blank?
+      return render(status: :bad_request, json: { message: 'Invalid event.' })
+    end
+
+    log_webhook_receipt
+    render json: { message: 'Secret token is valid.' }
```

Comment:
personally I'd go without explicit `return` statements?

also IMO none of the `render` calls should need parens, or all of them should have it, for consistency

```suggestion
    if !token_valid?
      render status: :unauthorized, json: { message: 'Invalid secret token.' }
    elsif socure_params[:event].blank?
      render status: :bad_request, json: { message: 'Invalid event.' }
    else
      log_webhook_receipt
      render json: { message: 'Secret token is valid.' }
    end
```

Reply:
Converted into before_actions